File: Problem513/Solution.go
package main

type TreeNode struct {
	Val   int
	Left  TreeNode
	Right TreeNode
}

var leftMostLvl int
var leftMostVal int

func findBottomLeftValue(root *TreeNode) int {
	leftMostVal = 0
	leftMostLvl = -1
	getLeftMostVal(root, 0)
	return leftMostVal
}

func getLeftMostVal(root *TreeNode, currLvl int) {
	if root == nil {
		return
	}
	if currLvl > leftMostLvl && root.Left == nil && root.Right == nil {
		leftMostLvl = currLvl
		leftMostVal = root.Val
	}
	getLeftMostVal(root.Left, currLvl+1)
	getLeftMostVal(root.Right, currLvl+1)
}

File: Problem513/Solution.java
public class Solution {
	private int leftMostLevel = 0;
	private int leftMostValue;

	public int findBottomLeftValue(TreeNode root) {
		getLeftMostVal(root, 0);
		return leftMostValue;
	}

	private void getLeftMostVal(TreeNode node, int currLvl) {
		if (node == null)
			return;
		if (leftMostLevel < currLvl && node.left == null && node.right == null) {
			leftMostLevel = currLvl;
			leftMostValue = node.val;
		}
		getLeftMostVal(node.left, currLvl + 1);
		getLeftMostVal(node.right, currLvl + 1);
	}
}